mpu6050/mpu6050_main.c

```diff
@@ -0,0 +1,373 @@
+		"i2c client address is 0x%X\n", drv_client->addr);
+
+	/* Read who_am_i register */
+	ret = i2c_smbus_read_byte_data(drv_client, REG_WHO_AM_I);
+	if (IS_ERR_VALUE(ret)) {
+		dev_err(&drv_client->dev,
+			"i2c_smbus_read_byte_data() failed with error: %d\n",
+			ret);
+		return ret;
+	}
+	if (ret != MPU6050_WHO_AM_I) {
+		dev_err(&drv_client->dev,
+			"wrong i2c device found: expected 0x%X, found 0x%X\n",
+			MPU6050_WHO_AM_I, ret);
+		return -1;
+	}
+	dev_info(&drv_client->dev,
+		"i2c mpu6050 device found, WHO_AM_I register value = 0x%X\n",
+		ret);
+
+	/* Setup the device */
+	/* No error handling here! */
+	i2c_smbus_write_byte_data(drv_client, REG_CONFIG, 0);
+	i2c_smbus_write_byte_data(drv_client, REG_GYRO_CONFIG, 0);
+	i2c_smbus_write_byte_data(drv_client, REG_ACCEL_CONFIG, 0);
+	i2c_smbus_write_byte_data(drv_client, REG_FIFO_EN, 0);
+	i2c_smbus_write_byte_data(drv_client, REG_INT_PIN_CFG, 0);
+	i2c_smbus_write_byte_data(drv_client, REG_INT_ENABLE, 0);
+	i2c_smbus_write_byte_data(drv_client, REG_USER_CTRL, 0);
+	i2c_smbus_write_byte_data(drv_client, REG_PWR_MGMT_1, 0);
+	i2c_smbus_write_byte_data(drv_client, REG_PWR_MGMT_2, 0);
+
+	get_mpu6050_data()->drv_client = drv_client;
+
+	dev_info(&drv_client->dev, "i2c driver probed\n");
+	return 0;
+}
+
+static int mpu6050_remove(struct i2c_client *drv_client)
+{
+	get_mpu6050_data()->drv_client = 0;
+
+	dev_info(&drv_client->dev, "i2c driver removed\n");
+	return 0;
+}
+
+static const struct i2c_device_id mpu6050_idtable[] = {
+	{ "mpu6050", 0 },
+	{ }
+};
+MODULE_DEVICE_TABLE(i2c, mpu6050_idtable);
+
+static struct i2c_driver mpu6050_i2c_driver = {
+	.driver = {
+		.name = "mpu6050",
+	},
+
+	.probe = mpu6050_probe,
+	.remove = mpu6050_remove,
+	.id_table = mpu6050_idtable,
+};
+
+static ssize_t data_show(struct kobject *kobj,
+				struct kobj_attribute *attr, char *buf)
+{
+	size_t index = get_attribute_index(attr);
+	struct mpu6050_data_elements current_element;
+	bool has_element;
+
+	has_element = get_active_element(get_mpu6050_data(), &current_element);
+
+	if (index < INDEX_COUNT && has_element)
+		sprintf(buf, "%d\n", current_element.data[index]);
+	else
+		buf[0] = '\0';
+	return strlen(buf);
+}
+
+static const struct kobj_attribute g_kobj_attributes[INDEX_COUNT] = {
+	__ATTR(accel_x, 0444, data_show, NULL),
+	__ATTR(accel_y, 0444, data_show, NULL),
+	__ATTR(accel_z, 0444, data_show, NULL),
+	__ATTR(gyro_x, 0444, data_show, NULL),
+	__ATTR(gyro_y, 0444, data_show, NULL),
+	__ATTR(gyro_z, 0444, data_show, NULL),
+	__ATTR(temperature, 0444, data_show, NULL)
+};
+
+static const struct attribute *g_attributes[INDEX_COUNT + 1] = {
+	&g_kobj_attributes[INDEX_ACCEL_X].attr,
+	&g_kobj_attributes[INDEX_ACCEL_Y].attr,
+	&g_kobj_attributes[INDEX_ACCEL_Z].attr,
+	&g_kobj_attributes[INDEX_GYRO_X].attr,
+	&g_kobj_attributes[INDEX_GYRO_Y].attr,
+	&g_kobj_attributes[INDEX_GYRO_Z].attr,
+	&g_kobj_attributes[INDEX_TEMPERATURE].attr,
+	NULL,
+};
+
+/* Gets a number of attributes + 1 (null-terminated) */
```

Comment:
In current implementation there's no need in `attributes_count()` function.
So why not to simplify code a bit?

Reply:
Done.